care/facility/api/viewsets/asset.py

```diff
@@ -412,6 +412,51 @@ def operate_assets(self, request, *args, **kwargs):
                 status=status.HTTP_500_INTERNAL_SERVER_ERROR,
             )
 
+    @extend_schema(tags=["asset"])
+    @action(methods=["POST"], detail=False)
+    def bulk_upsert(self, request, *args, **kwargs):
+        if "assets" not in request.data:
+            raise ValidationError({"assets": "No Data was provided"})
+        if not isinstance(request.data["assets"], list):
+            raise ValidationError({"assets": "Invalid Data"})
+        if "location" not in request.data:
+            raise ValidationError({"location": "Location is required"})
+
+        limit = 200
+
+        if len(request.data["assets"]) > limit:
+            raise ValidationError(
+                {"assets": f"Maximum of {limit} assets can be created at once"}
+            )
+
+        assets = request.data.get("assets", [])
+        location = request.data.get("location")
+        errors = []
+        counter = 0
+        ser_objects = []
+        invalid = False
+        for asset in assets:
+            counter += 1
+            asset["location"] = location
+            serialiser_obj = AssetSerializer(data=asset, context={"request": request})
+            valid = serialiser_obj.is_valid()
+            current_error = serialiser_obj.errors
+            if current_error and (not valid):
+                errors.append({"index": counter, "error": current_error})
+                invalid = True
+            ser_objects.append(serialiser_obj)
+
+        if invalid:
+            return Response(errors, status=status.HTTP_400_BAD_REQUEST)
+
+        for ser in ser_objects:
+            ser.save()
```

Comment:
use bulk create

Reply:
getting some issues in using bulk create

Reply:
what issues exactly?

Reply:
![image](https://github.com/coronasafe/care/assets/111575806/bce06f74-c6e3-419c-9086-4dca1a99c6e1)
![image](https://github.com/coronasafe/care/assets/111575806/a572213b-6ffd-4df3-b5b4-5767eef29181)

Reply:
The problem is with the fields 'note'and 'last_serviced_on' fields , 
![image](https://github.com/coronasafe/care/assets/111575806/b1abb2d8-740c-445d-a548-9b2acf0ab0bb)
throws an error :
![image](https://github.com/coronasafe/care/assets/111575806/25425ab3-f07f-481b-84c4-4fb9bd5d2f19)

how ever this worked 
![image](https://github.com/coronasafe/care/assets/111575806/79b35d1f-19a1-4a28-9f63-71a834d9af2a)
i m not sure which would be a better approch